include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -231,6 +231,36 @@ struct Session : Base<OrtSession> {
   TypeInfo GetOverridableInitializerTypeInfo(size_t index) const;
 };
 
+/**
+ * An experimental wrapper around the C++ API that provides support for modern C++ syntax/features
+ *   at the cost of some (minimal) memory overhead.
+ * Where possible default values are defined by the session's model metadata (i.e. input node names)
+ *   unless otherwise specified by the user.
+ * 
+ * NOTE: Experimental API components provide no guarantee of backwards compatibility in future releases.
+ */
+struct ExperimentalSession : Session {
+  ExperimentalSession(Env& env, ORTCHAR_T* model_path, SessionOptions& options) 
+    : Session(env, model_path, options) {};
```

Comment:
Can this support a std::string for the model_path and internally convert that to ORTCHAR_T?

Reply:
The you need to know the encoding, which will be a much bigger topic.